code/modules/projectiles/firing.dm

```diff
@@ -1,8 +1,8 @@
-/obj/item/ammo_casing/proc/fire(atom/target as mob|obj|turf, mob/living/user as mob|obj, params, distro, quiet, zone_override = "", spread, obj/firer_source_object)
+/obj/item/ammo_casing/proc/fire(atom/target as mob|obj|turf, mob/living/user as mob|obj, params, distro, quiet, zone_override = "", spread, atom/firer_source_atom)
```

Comment:
Would you remove the `as ...` stuff from the parameters?

Reply:
done